pandas/core/generic.py

```diff
@@ -5134,12 +5134,15 @@ def tail(self: FrameOrSeries, n: int = 5) -> FrameOrSeries:
     @final
     def sample(
         self: FrameOrSeries,
-        n=None,
-        frac=None,
-        replace=False,
-        weights=None,
-        random_state=None,
-        axis=None,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool_t = False,
+        weights: Union[str, np.ndarray] = None,
```

Comment:
needs to be `'Optional`

Also could this be an ExtensionArray (in which case use `ArrayLike`)

Reply:
Thanks, fixed it in last commit.